.github/workflows/ci-nodejs.yml

```diff
@@ -0,0 +1,69 @@
+name: CI (Node.js)
+
+on:
+  workflow_call:
+    inputs:
+      node-version:
+        type: number
+        required: true
+    secrets:
+      aws-id:
+        required: true
+      aws-secret:
+        required: true
+      git-read-key:
+        required: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.aws-id }}
+          aws-secret-access-key: ${{ secrets.aws-secret }}
+          aws-region: eu-central-1
+
+      - name: Install SSH key
+        uses: webfactory/ssh-agent@v0.5.4
+        with:
+          ssh-private-key: ${{ secrets.git-read-key }}
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ inputs.node-version }}
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Test
+        run: npm run test --if-present
+
+  lint:
```

Comment:
I think you shouldn't use two jobs. Lint is super quick but you run npm ci a few times slower.